### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by ArgoNavis1763**
All functionality for authentication working and all tests passing

### CODE REVIEW

File: cypress.env.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "auth_audience": "https://dev-f3mw0pd9.eu.auth0.com/api/v2/",
+  "auth_url": "https://dev-f3mw0pd9.eu.auth0.com/oauth/token",
+  "auth_client_id": "VeOpwVfDQqX7gGRB2H8g2JuE4EvpGe6y",
+  "auth_client_secret": "x3dVW6BaPYGdgr6aowxFqpj_J2TxzXx6cAKfRqKdlq9LVcM7fLXkuDbzJjbRnpMu",
+  "auth_username": "carina.druce@zopa.com",
```
Comment (TomGroombridge) [inline cypress.env.json]: I think this file should be in your .gitignore file as we don't want to commit these details

File: cypress/integration/comments-section/actioning-comments.spec.js [hunk 2]
```diff
@@ -1,11 +1,32 @@
...
   beforeEach(() => {
+    cy.visit('http://localhost:3000');
     cy.fixture('comments-response.json').then((json) => {
       cy.intercept('GET', 'https://api.mocki.io/v2/a20ae30b/comments', json);
     });
-    cy.visit('http://localhost:3000');
+    cy.get('#login-button').click();
   });
 
+  // beforeEach(() => {
```
Comment (TomGroombridge) [inline cypress/integration/comments-section/actioning-comments.spec.js]: remove code that's commented out

File: cypress/integration/comments-section/login.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// describe('login', () => {
```
Comment (TomGroombridge) [inline cypress/integration/comments-section/login.js]: delete this file

File: src/components/comment.jsx [hunk 4]
```diff
@@ -27,22 +29,28 @@ const Comment = (props) => {
...
       }
       return comment;
     });
+    console.log('replyData', replyData);
 
+    // console.log('props.reply.user.username', props.reply.user.username);
     setComments(newComments);
     setReplyClicked(false);
     setReplyValue('');
+    // setCurrentUser(user);
```
Comment (TomGroombridge) [inline src/components/comment.jsx]: remove consoles and comments

File: src/components/edit.jsx [hunk 5]
```diff
@@ -18,15 +15,17 @@ const Edit = (props) => {
     }
 
     const newComments = [...comments];
+    console.log('index', index);
     newComments[index] = { ...comment, content: content };
     setComments(newComments);
     setEditClicked(false);
+    console.log('newComments', newComments);
```
Comment (TomGroombridge) [inline src/components/edit.jsx]: consoles

File: src/pages/Comments/index.jsx [hunk 6]
```diff
@@ -54,48 +61,85 @@ const Comments = () => {
...
     setAddedComment('');
+    console.log('nickname', user.nickname);
   };
 
-  if (loading) return <h1>Loading...</h1>;
+  if (loading || isLoading) return <h1>Loading...</h1>;
 
   if (error) return <h1>There has been an error</h1>;
-
+  // console.log(user);
```
Comment (TomGroombridge) [inline src/pages/Comments/index.jsx]: consoles in this file

File: tests.env [hunk 7]
```diff
@@ -0,0 +1,2 @@
+USERNAME: "carina.druce@zopa.com",
```
Comment (TomGroombridge) [inline tests.env]: what's this file for?

File: cypress/integration/comments-section/actioning-comments.spec.js (hunk 2 above)
Comment (ArgoNavis1763) [reply cypress/integration/comments-section/actioning-comments.spec.js]: done

File: cypress/integration/comments-section/login.js (hunk 3 above)
Comment (ArgoNavis1763) [reply cypress/integration/comments-section/login.js]: done

File: src/components/comment.jsx (hunk 4 above)
Comment (ArgoNavis1763) [reply src/components/comment.jsx]: done

File: src/components/edit.jsx (hunk 5 above)
Comment (ArgoNavis1763) [reply src/components/edit.jsx]: done

File: src/pages/Comments/index.jsx (hunk 6 above)
Comment (ArgoNavis1763) [reply src/pages/Comments/index.jsx]: done

File: tests.env (hunk 7 above)
Comment (ArgoNavis1763) [reply tests.env]: We were gonna use it to not share email and password but they're not real so it's been deleted and file will be added to gitignore

File: cypress.env.json (hunk 1 above)
Comment (ArgoNavis1763) [reply cypress.env.json]: Added to .env

**Pull request merged**